decisions/DR-update-devkit-architecture.md

```diff
@@ -0,0 +1,63 @@
+# Update devkit architecture
+
+| Status      | draft |
+| :---------- | :-------------------------------------------------------------------------------------- |
+| **PR #**    | []() |
+| **Author & Sponsor** | [Marko](https://github.com/m4rk055) (marko@blindnet.io) |
+| **Reviewers** | [Milan](https://github.com/milstan), [Amaury](https://github.com/TheKinrar), [Justin](https://github.com/jboileau99) |
+| **Assignee** | [Milan](https://github.com/milstan) |
+
+## Context and Problem Statement
+
+The model described in the [PRIV](https://github.com/blindnet-io/product-management/blob/devkit-schemas/refs/schemas/priv/RFC-PRIV.md) document requires the [Privacy Compiler](https://github.com/blindnet-io/product-management/tree/main/refs/high-level-architecture#privacy-compiler) to be aware of every [Data Captures and Capture Fragments](https://github.com/blindnet-io/product-management/tree/main/refs/high-level-architecture#privacy-compiler) in the system.
+That puts a burden on the implementing system to
+- make sure every data point collected by the system is as well referenced (metadata) in the Privacy Computation Engine database
+- collect or transform data to a particular format (capture and fragments)
+
+This DR proposes removing the notion of Data Captures and Capture Fragments from Privacy Computation Engine.
+
+## Proposal
+
+The [Privacy computation engine](https://github.com/blindnet-io/product-management/tree/main/refs/high-level-architecture#privacy-computation-engine) (PCE) should have the following components:
+- Privacy Compiler
+- Privacy Request Capture Interface
+- Data Consumer/DPO Interface
+- Data access engine
+- Data storage connectors (for e.g. Postgres, MongoDB, Azure Storage)
+
+**Data access engine** is a component deployed by a client. It communicates with the **Privacy Compiler** and a **Data storage connector**.
```

Comment:
Seems to me like the Data access engine serves no purpose other than relaying requests from compiler to storage connector, we could probably remove it?

Or, if we want it to have some important roles, we should probably specify it. I see how it might be useful to have it as a common interface for all parts of the Privacy engine (and even whole devkit) that prevents having to directly communicate with the storage connector, but I'm not sure, the storage connectors must implement a common API that we define.

Reply:
It might do some additional work like scheduling tasks for the connector. I'll think about the complete scope of it's operations and if it's useful to have it as a separate component.

Reply:
with regards to Data Access Engine there is also a need (that we have to accoutn for) when replying to an ACCESS request, to fetch and display data to the Data Subject.

@TheKinrar @m4rk055 I think we simply, now, have insufficient information on how to best organise the the connection with client storage. The best way to do it will become clear as we start developing, so I suggest we settle here on a temporary, good enough for now definition, and then we'll update later.

Reply:
What we have here is good enough in terms of precision, we can use the other DR in #748 to then define the technical specifics as we previously did. This is not about how it actually works, only about the roles of the components. This is why I still believe it's useful to know if a component is needed